pkg/tagger/tagstore/tagstore.go

```diff
@@ -338,30 +360,66 @@ type tagPriority struct {
 	cardinality collectors.TagCardinality    // cardinality level of the tag (low, orchestrator, high)
 }
 
-func (e *entityTags) get(cardinality collectors.TagCardinality) ([]string, []string) {
+func (e *EntityTags) get(cardinality collectors.TagCardinality) ([]string, []string) {
+	tags, sources := e.getBuilder(cardinality)
+	return tags.Get(), sources
+}
+
+func (e *EntityTags) getBuilder(cardinality collectors.TagCardinality) (*util.HashedTags, []string) {
 	e.computeCache()
 
 	if cardinality == collectors.HighCardinality {
-		return e.cachedAll, e.cachedSource
+		return &e.cachedAll, e.cachedSource
 	} else if cardinality == collectors.OrchestratorCardinality {
-		return e.cachedOrchestrator, e.cachedSource
+		return &e.cachedOrchestrator, e.cachedSource
 	}
-	return e.cachedLow, e.cachedSource
+	return &e.cachedLow, e.cachedSource
 }
 
-func (e *entityTags) toEntity() types.Entity {
+func (e *EntityTags) toEntity() types.Entity {
 	e.computeCache()
 
+	cachedAll := e.cachedAll.Get()
+	cachedOrchestrator := e.cachedOrchestrator.Get()
+	cachedLow := e.cachedLow.Get()
+
 	return types.Entity{
 		ID:                          e.entityID,
 		StandardTags:                e.getStandard(),
-		HighCardinalityTags:         e.cachedAll[len(e.cachedOrchestrator):],
-		OrchestratorCardinalityTags: e.cachedOrchestrator[len(e.cachedLow):],
-		LowCardinalityTags:          e.cachedLow,
+		HighCardinalityTags:         cachedAll[len(cachedOrchestrator):],
+		OrchestratorCardinalityTags: cachedOrchestrator[len(cachedLow):],
+		LowCardinalityTags:          cachedLow,
```

Comment:
It skips orchestrator tags instead of selecting them; I added some comments above (https://github.com/DataDog/datadog-agent/commit/c43482705bc846084f779a150c66e502d73d1d0e)